client/payment-details/payment-method/klarna/index.js

```diff
@@ -0,0 +1,146 @@
+/** @format **/
+
+/**
+ * External dependencies
+ */
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies.
+ */
+import PaymentDetailsPaymentMethodDetail from '../detail';
+
+/**
+ * Extracts and formats payment method details from a charge.
+ *
+ * @param {Object} charge The charge object.
+ * @return {Object}       A flat hash of all necessary values.
+ */
+const formatPaymentMethodDetails = ( charge ) => {
+	const { billing_details: billingDetails, payment_method: id } = charge;
+
+	const {
+		country: countryCode,
+		verified_name: verifiedName,
+		payment_method_category: paymentMethodCategory,
+		preferred_locale: preferredLocale,
+	} = charge.payment_method_details.klarna;
+
+	const { name, email, formatted_address: formattedAddress } = billingDetails;
+	// Use the full country name.
+	const country = wcSettings.countries[ countryCode ];
+
+	return {
+		verifiedName,
+		id,
+		name,
+		email,
+		country,
+		formattedAddress,
+		paymentMethodCategory,
+		preferredLocale,
+	};
+};
+
+/**
+ * Placeholders to display while loading.
+ */
+const paymentMethodPlaceholders = {
+	verifiedName: 'verified name placeholder',
+	fingerprint: 'fingerprint placeholder',
+	id: 'id placeholder',
+	name: 'name placeholder',
+	email: 'email placeholder',
+	formattedAddress: 'address placeholder',
+	country: 'country placeholder',
+};
+
+const KlarnaDetails = ( { charge = {}, isLoading } ) => {
+	const details = charge.payment_method_details
+		? formatPaymentMethodDetails( charge )
+		: paymentMethodPlaceholders;
+
+	const {
+		verifiedName,
+		id,
+		name,
+		email,
+		country,
+		formattedAddress,
+		paymentMethodCategory,
+		preferredLocale,
+	} = details;
+
+	// Shorthand for more readable code.
+	const Detail = PaymentDetailsPaymentMethodDetail;
+
+	return (
+		<div className="payment-method-details">
+			<div className="payment-method-details__column">
+				<Detail
+					isLoading={ isLoading }
+					label={ __( 'ID', 'woocommerce-payments' ) }
+				>
+					{ id }
+				</Detail>
+
+				<Detail
+					isLoading={ isLoading }
+					label={ __( 'Category', 'woocommerce-payments' ) }
+				>
+					{ paymentMethodCategory }
+				</Detail>
```

Comment:
Nice touch! I wonder if we should add mapping somewhere to show possible values of `pay_later`, `pay_now`, `pay_with_financing`, or `pay_in_installments` as a readable/translatable text?

Reply:
We should probably TS it. But I'm not sure about translating them. At the end of the day, those are codes not messages. If merchant tries to figure out what it means, and it's translated, then it will be hard to Google it.

Reply:
Stripe translates those in their dashboard, I don't think we should hesitate to do the same.